sshuttle/client.py

```diff
@@ -563,10 +566,9 @@ def tcp_connection_is_allowed_conditional(dstip, dstport, srcip, check_acl, chec
             debug3("Connection from a source excluded from the ACL\n")
             return True
 
-        # TODO: eventually this will check aclAlwaysConnected to verify workspace IPs
         check_allowed_sources = True
-        if _always_connected == ALWAYS_CONNECTED_ON:
-            debug1("TCP connection source is allowed because alwaysConnected mode is ON\n")
+        if (_always_connected == ALWAYS_CONNECTED_ON) and (srcip in _acl_always_connected):
+            debug3("TCP source allowed because alwaysConnected mode is ON and srcip is in aclAlwaysConnected\n")
```

Comment:
Does value of srcip show up in debug log?

Reply:
no, but I can add it if it helps debugging

Reply:
It could help...